CMakeLists.txt

```diff
@@ -86,89 +86,74 @@ if(NOT CUDA_FOUND)
   message(FATAL_ERROR "Could not find CUDA >= 7.0")
 endif()
 
-#
-# Default setting of the CUDA CC versions to compile.
-# Shortening the lists saves a lot of compile time.
-#
-if(CUDA_VERSION_MAJOR GREATER 7)
-  set(PopSift_CUDA_CC_LIST_BASIC 30 35 50 52 60 61 62)
+message(STATUS "CUDA Version is ${CUDA_VERSION}")
+
+include(ChooseCudaCC)
+if(NOT DEFINED PopSift_CUDA_CC_LIST)
+  chooseCudaCC(PopSift_CUDA_CC_LIST_BASIC
+               PopSift_CUDA_GENCODE_FLAGS
+	       MIN_CC 30
+	       MIN_CUDA_VERSION 7.0)
+  set(PopSift_CUDA_CC_LIST ${PopSift_CUDA_CC_LIST_BASIC} CACHE STRING "CUDA CC versions to compile")
 else()
-  set(PopSift_CUDA_CC_LIST_BASIC 30 35 50 52 )
+  getFlagsForCudaCCList(PopSift_CUDA_CC_LIST
```

Comment:
So I guess we want to give the possibility to the user to provide a `-DPopSift_CUDA_CC_LIST=30;70` at cmake stage, am I correct?
In that case:

1. we should document it in the README in the build section

2. more importantly, we should check that those provided are good/supported. One way to go it could be something like:
```cmake
# we always need to know which are all the supported CC
chooseCudaCC(PopSift_CUDA_CC_LIST_BASIC 
            PopSift_CUDA_GENCODE_FLAGS
            MIN_CC 30
            MIN_CUDA_VERSION 7.0)
# if PopSift_CUDA_CC_LIST is provided
if(DEFINED PopSift_CUDA_CC_LIST)
    # check each provided element is supported
    # ideally this function checks the provided cc are in the list generating an error if not
    check_if_supported(PopSift_CUDA_CC_LIST  PopSift_CUDA_CC_LIST_BASIC)
    getFlagsForCudaCCList(PopSift_CUDA_CC_LIST PopSift_CUDA_GENCODE_FLAGS)
else()
    set(PopSift_CUDA_CC_LIST ${PopSift_CUDA_CC_LIST_BASIC} CACHE STRING "CUDA CC versions to compile")
endif()
list(APPEND CUDA_NVCC_FLAGS "${PopSift_CUDA_GENCODE_FLAGS}")
```
and
```cmake
function(check_if_supported PROVIDED_CC  SUPPORTED_CC)
    foreach(cc IN LISTS PROVIDED_CC)
        if(NOT ${cc} IN_LIST SUPPORTED_CC)
            message(FATAL_ERROR "Compute capability ${cc} not supported. Supported CC are ${SUPPORTED_CC}")
    endforeach()
endfunction()
```

What do you think?

Reply:
Is that a really good thing? Without that check, a user can configure for a brand new CC that we haven't put into the ChooseCudaCC.cmake file yet.

Reply:
Well, you have a point! 
We can then leave it like that and assume that it is an advanced feature and the user knows what is doing.